### ISSUE TITLE
Error importing AWS targetGroup with pulumi cli

### ISSUE THREAD

**Issue by SergioGenially** [pulumi/pulumi-aws#2534]
### What happened?

I encountered an error while executing the import command for AWS targetGroup:

`pulumi import aws:alb/targetGroup:TargetGroup new-targetgroup arn:aws:elasticloadbalancing:eu-west-1:XXXXXXXX:targetgroup/my_targetGroup/YYYYYYYYYY`

The error message received was:

`error: anonymous.pp:28,22-29,14: cannot assign expression of type (null) to location of type list({ onDeregistration: output(string) | string, onUnhealthy: output(string) | string } | output({ onDeregistration: string, onUnhealthy: string })) | output(list({ onDeregistration: string, onUnhealthy: string }))?: ;`

### Expected Behavior

The resource TargetGroup should have been imported correctly without any error.

### Steps to reproduce

To reproduce the issue, just execute the targetGroup import command:

pulumi import aws:alb/targetGroup:TargetGroup 

### Output of `pulumi about`

```
pulumi about
CLI          
Version      3.68.0
Go Version   go1.20.4
Go Compiler  gc

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

Backend        
Name           dom
URL            s3://pulumi-state-backends
User           sergio
Organizations  
```

### Additional context

I have attempted to downgrade the Pulumi CLI to previous versions, but I am still encountering the same error.

### Contributing

I would like to contribute to the resolution of this issue. Please let me know if there is any guidance or assistance I can provide.

Thank you for your attention to this matter.

**Comment by MoralGr** [comment]
It happens exactly the same.
A month ago I imported a target group and it worked, but currently I get that error

**Comment by mikhailshilkov** [comment]
@SergioGenially and @MoralGr  Thank you for reporting this issue.

I'm trying to reproduce this with all the latest: CLI 3.68.0 and @pulumi/aws 5.41.0. I created a target group with the following repro:

```ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const main = new aws.ec2.Vpc("main", {cidrBlock: "10.0.0.0/16"});
const test = new aws.lb.TargetGroup("test", {
    port: 80,
    protocol: "HTTP",
    vpcId: main.id,
});
```

And then import the resulting target group as another resource. The command succeeds for me.

I suspect I need to set some other property on my target group to reproduce. Could you please provide more details about your target group configuration, ideally, as a Pulumi program?

Also, which version of the provider are you using?

Thank you!

**Comment by SergioGenially** [comment]
Hello @mikhailshilkov.

We was using @pulumi/aws 5.30.0... After upgrading the plugin to 5.41.0 the import ran successfully.

Thank you so much for your support and sorry for the inconvenience.

**Comment by MitchellGerdisch** [comment]
This issue has reappeared starting with v6.7.0 and confirmed in 6.13.3.
The issue does not exist in v6.6.1.

**Comment by t0yv0** [comment]
Noting https://github.com/pulumi/pulumi/blob/master/pkg/codegen/hcl2/model/diagnostics.go#L41 possibly where the error is coming from.

**Comment by t0yv0** [comment]
Thank you all, apologies for the experience here, we'll have a look shortly, this sounds like it's something that need to be fixed by an integration test so we stop regressing on it.

**Comment by t0yv0** [comment]
https://github.com/pulumi/pulumi-aws/issues/2517 I believe is where this was dealt with before.

**Comment by t0yv0** [comment]
6.32.0 - not affected
6.21.0 - not affected
6.18.2 - not affected
6.18.1 - affected
6.18.0 - affected
6.17.0 - affected
6.13.3 - affected

**Comment by t0yv0** [comment]
Adding a regression test in https://github.com/pulumi/pulumi-aws/pull/3859 and closing for now as it appears to be fixed in the latest provider v6.32.0.
